nbgrader/docs/source/user_guide/tests.yml

```
@@ -0,0 +1,305 @@
+python3:
+    setup: "from hashlib import sha1"
+    hash: 'sha1({{snippet}}.encode("utf-8")+b"{{salt}}").hexdigest()'
```

Don't you think this could be something like:
```suggestion
    hash:
        setup: "from hashlib import sha1"
        command: 'sha1({{snippet}}.encode("utf-8")+b"{{salt}}").hexdigest()'
```
to have a special setup only for hash, keeping also a general setup for other configuration than hash.
This would allow importing the hash function only if hash is required.

---

I personally don't think it's worth the change--the `setup` is general anyway, it can be used for whatever. I don't feel strongly, though, so if you feel strongly I can change it. Just let me know